### PR TITLE
vmware.py VM creation with Datastore Cluster

### PR DESCRIPTION
**Pull request by preussal**
VM creation with Datastore Cluster
When cloning a template that resides in a datastore cluster, that already works.
With the customization, you can create a new VM without a template on a Datasore cluster.

What does this PR do?
VM creation with Datastore Cluster

What issues does this PR fix or reference?
Cannot create a new VM on Datasore Cluster, only on Datastore
When cloning a template that resides in a datastore cluster, that already works.

Previous Behavior
Cannot create a new VM on Datasore Cluster, only on Datastore

New Behavior
With the customization, you can create a new VM without a template on a Datasore Cluster.

Tests written?
No
Test only on my enviorment

Commits signed with GPG?
No

Please review Salt's Contributing Guide for best practices.

See GitHub's page on GPG signing for more information about signing commits with GPG.

### CODE REVIEW

**Comment by xeacott** [comment]
Thanks for getting this together! I see that the lint tests are failing, those will need to get fixed and I'll also leave some code comments. And finally, since this PR introduces new logic I'm going to ask if you could put some tests together for this, too. We actually have a lot of examples in [here](https://github.com/saltstack/salt/blob/develop/tests/unit/modules/test_vsphere.py).

File: salt/cloud/clouds/vmware.py [hunk 1]
```diff
@@ -2822,7 +2822,33 @@ def create(vm_):
...
+                         raise SaltCloudSystemExit("Specified datastore: '{0}' does not exist".format(datastore))
+
+                 # datastore cluster has been specified
+                 # find datastore with most free space available
+                 #
+                 # TODO: Get DRS Recommendations instead of finding datastore with most free space
+                 datastore_list = salt.utils.vmware.get_datastores(_get_si(), datastore_cluster_ref, get_all_datastores=True)
+                 datastore_free_space = 0
+                 for ds_ref in datastore_list:
+                     log.trace(
```
Comment (xeacott) [inline salt/cloud/clouds/vmware.py]: Is this appropriate to add a log.trace here? Versus a log.info or log.msg?

**Review by xeacott — CHANGES_REQUESTED** [review]

**Comment by dwoz** [comment]
@preussal We are no longer accepting PRs to the `develop` branch. This needs to be re-opened against the `master` branch. You can link to this PR from the new one to preserve any disscussion history. Sorry for the confusion.

**Comment by sagetherage** [comment]
closing as abandoned

**Pull request closed without merging**